src/mlpack/bindings/go/get_printable_type_impl.hpp

```diff
@@ -0,0 +1,142 @@
+    const typename boost::disable_if<arma::is_arma_type<double>>::type*,
+    const typename boost::disable_if<std::is_same<double,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "float64";
+}
+
+template<>
+inline std::string GetPrintableType<std::string>(
+    const util::ParamData& /* d */,
+    const typename boost::disable_if<util::IsStdVector<std::string>>::type*,
+    const typename boost::disable_if<data::HasSerialize<std::string>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<std::string>>::type*,
+    const typename boost::disable_if<std::is_same<std::string,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "string";
+}
+
+template<>
+inline std::string GetPrintableType<bool>(
+    const util::ParamData& /* d */,
+    const typename boost::disable_if<util::IsStdVector<bool>>::type*,
+    const typename boost::disable_if<data::HasSerialize<bool>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<bool>>::type*,
+    const typename boost::disable_if<std::is_same<bool,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "bool";
+}
+
+template<typename T>
+inline std::string GetPrintableType(
+    const util::ParamData& d,
+    const typename boost::enable_if<util::IsStdVector<T>>::type*,
+    const typename boost::disable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "array of " + GetPrintableType<typename T::value_type>(d) + "s";
+}
+
+template<typename T>
+inline std::string GetPrintableType(
+    const util::ParamData& /* d */,
+    const typename boost::enable_if<arma::is_arma_type<T>>::type*,
+    const typename boost::disable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  std::string type = "*mat.Dense";
+  if (std::is_same<typename T::elem_type, double>::value)
+  {
+    if (T::is_row || T::is_col)
+      type = "*mat.Dense (1d)";
+  }
+  else if (std::is_same<typename T::elem_type, size_t>::value)
+  {
+    type = "*mat.Dense (with ints)";
+    if (T::is_row || T::is_col)
+      type = "*mat.Dense (1d with ints)";
+  }
+
+  return type;
+}
+
+template<typename T>
+inline std::string GetPrintableType(
+    const util::ParamData& /* d */,
+    const typename boost::enable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  return "matrixWithInfo";
+}
+
+template<typename T>
+inline std::string GetPrintableType(
+    const util::ParamData& d,
+    const typename boost::disable_if<arma::is_arma_type<T>>::type*,
+    const typename boost::enable_if<data::HasSerialize<T>>::type*,
+    const typename boost::disable_if<std::is_same<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>::type*)
+{
+  std::string goStrippedType, strippedType, printedType, defaultsType;
+  StripType(d.cppType, goStrippedType, strippedType, printedType, defaultsType);
+  return goStrippedType + "Type";
```

Comment:
Are you sure the `Type` is needed here?  The documentation prints `adaboostModelType`: http://mlpack.ratml.org/doc/mlpack-git/go_documentation.html#adaboost for `InputModel`, but the code doesn't have that type:

```go
type adaBoostModel struct {
  mem unsafe.Pointer
}
```

Reply:
Updated.